src/clarity-angular/datagrid/render/column-resizer.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Directive, ElementRef, Renderer, OnDestroy, AfterViewInit, Output, EventEmitter} from "@angular/core";
+import {DomAdapter} from "./dom-adapter";
+import {DatagridRenderOrganizer} from "./render-organizer";
+
+@Directive({
+    selector: "clr-dg-column"
+})
+export class DatagridColumnResizer implements AfterViewInit, OnDestroy {
+
+
+    constructor(private el: ElementRef,
+                private renderer: Renderer,
+                private organizer: DatagridRenderOrganizer,
+                private domAdapter: DomAdapter) {
+
+        this.columnEl = el.nativeElement;
+    }
+
+    columnEl: Element;
+    columnRectWidth: number;
+    columnResizeBy: number = 0;
+
+    draggerEl: Element;
+    draggerElWidth: number;
+    draggerTracker: Element;
+
+    pageStartPositionX: number;
+    dragDistancePositionX: number; //relative to pageStartPosition
+
+    @Output("clrDgColumnResize") resizeEmitter: EventEmitter<any> = new EventEmitter();
+
+    dragWithinMinWidth: boolean = false;
+
+    private _columnMinWidth: number;
+
+    get columnMinWidth(): number {
+        return this._columnMinWidth;
+    }
+
+    set columnMinWidth(value: number) {
+        this._columnMinWidth = value || 96;
+    }
+
+    private columnResizeEvents: Function[];
+
+    ngAfterViewInit() {
+        this.columnResizeEvents = this.addDragListener();
+    }
+
+    ngOnDestroy() {
+        this.columnResizeEvents.map(event => event());
+    }
+
+
+    addDragListener(): Function[] {
+
+        this.draggerEl = this.columnEl.querySelector("button.separator-dragger");
+        this.draggerTracker = this.columnEl.querySelector(".separator-tracker");
+
+        this.draggerElWidth = this.domAdapter.clientRectWidth(this.draggerEl);
+
+        this.columnMinWidth = this.domAdapter.minWidth(this.columnEl);
+
+        return this.listenDrag();
+    }
+
+    listenDrag(): Function[] {
+
+        let mouseDownListener: any;
+        let mouseMoveListener: any;
+        let mouseUpListener: any;
+
+        let touchStartListener: any;
+        let touchMoveListener: any;
+        let touchEndListener: any;
+
+        mouseDownListener = this.renderer.listen(this.draggerEl, "mousedown", () => {
+            this.onDragStart();
+
+            mouseMoveListener = this.renderer.listenGlobal("document", "mousemove", (mouseMoveEvent: any) => {
```

Comment:
I think we might want to run this listener outside of the NgZone. At least we should try.